src/main/java/com/nvidia/spark/rapids/jni/kudo/KudoTableMerger.java

```diff
@@ -155,80 +141,64 @@ private static int copyValidityBuffer(HostMemoryBuffer dest, int startBit,
                                         HostMemoryBuffer src, int srcOffset,
                                         SliceInfo sliceInfo) {
     int nullCount = 0;
-    int totalRowCount = sliceInfo.getRowCount();
-    int curIdx = 0;
-    int curSrcByteIdx = srcOffset;
-    int curSrcBitIdx = sliceInfo.getValidityBufferInfo().getBeginBit();
-    int curDestByteIdx = startBit / 8;
-    int curDestBitIdx = startBit % 8;
-
-    while (curIdx < totalRowCount) {
-      int leftRowCount = totalRowCount - curIdx;
-      int appendCount;
-      if (curDestBitIdx == 0) {
-        appendCount = min(8, leftRowCount);
-      } else {
-        appendCount = min(8 - curDestBitIdx, leftRowCount);
-      }
-
-      int leftBitsInCurSrcByte = 8 - curSrcBitIdx;
-      byte srcByte = src.getByte(curSrcByteIdx);
-      if (leftBitsInCurSrcByte >= appendCount) {
-        // Extract appendCount bits from srcByte, starting from curSrcBitIdx
-        byte mask = (byte) (((1 << appendCount) - 1) & 0xFF);
-        srcByte = (byte) ((srcByte >>> curSrcBitIdx) & mask);
+    int totalRowCount = toIntExact(sliceInfo.getRowCount() + sliceInfo.getValidityBufferInfo().getBeginBit());
+    int curSrcIdx = sliceInfo.getValidityBufferInfo().getBeginBit();
+    int curDestIdx = startBit;
 
-        nullCount += (appendCount - NUMBER_OF_ONES[srcByte & 0xFF]);
 
-        // Sets the bits in destination buffer starting from curDestBitIdx to 0
-        byte destByte = dest.getByte(curDestByteIdx);
-        destByte = (byte) (destByte & ((1 << curDestBitIdx) - 1) & 0xFF);
+    while (curSrcIdx < totalRowCount) {
+      int leftRowCount = totalRowCount - curSrcIdx;
 
-        // Update destination byte with the bits from source byte
-        destByte = (byte) ((destByte | (srcByte << curDestBitIdx)) & 0xFF);
-        dest.setByte(curDestByteIdx, destByte);
+      int curDestOffset = (curDestIdx / 32) * Integer.BYTES;
```

Comment:
consider using mnemonic constants 
```suggestion
      int curDestOffset = (curDestIdx / Integer.SIZE) * Integer.BYTES;
```
or even 
```suggestion
      int curDestOffset = curDestIdx / Byte.SIZE;
```